src/utils/members.ts

```diff
@@ -22,3 +27,38 @@ export const getColonyContributorId = (
   colonyAddress: string,
   walletAddress: string,
 ) => `${colonyAddress}_${walletAddress}`;
+
+export const getColonyContributorAddressFromId = (
+  colonyContributorId: string,
+) => colonyContributorId.split('_')[1];
+
+export const invalidateMemberQueries = async (
+  userAddresses: string[],
+  colonyAddress: string,
+): Promise<void> => {
+  await apolloClient.refetchQueries({
+    include: [GetColonyContributorsDocument],
+  });
+  await apolloClient.refetchQueries({
+    include: [GetColonyContributorDocument],
+    onQueryUpdated: (query) => {
+      // useGetColonyContributorQuery variables
+      if (
+        !query.variables?.id ||
+        !query.variables?.colonyAddress ||
+        query.variables.colonyAddress !== colonyAddress
+      ) {
+        return false;
+      }
+
+      const colonyContributorId = query.variables.id;
+
+      const contributorAddress =
+        getColonyContributorAddressFromId(colonyContributorId);
+      const isUserAffected = userAddresses.indexOf(contributorAddress) !== -1;
+
+      // refetch only if user's verified status was changed
+      return isUserAffected;
+    },
+  });
+};
```

Comment:
Can you explain why we need to call refetchQueries twice, preferably as an in-code comment?

Reply:
I'll see if I can join them into one, but essentially the second one uses the `onQueryUpdated` callback, which should trigger just for the `GetColonyContributor` query and not any other ones. Maybe I can do a name check in the first line, nice catch